zebra-consensus/src/block/tests.rs

```diff
@@ -4,181 +4,115 @@ use super::*;
-
-    // Get a valid block
-    let mut block = Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
-        .expect("block test vector should deserialize");
-
-    // This should be ok
-    ready_verifier_service
-        .call(Arc::new(block.clone()))
-        .await
-        .map_err(|e| eyre!(e))?;
+use zebra_test::transcript::{TransError, Transcript};
+
+static VALID_BLOCK_TRANSCRIPT: Lazy<Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>> =
+    Lazy::new(|| {
+        let block: Arc<_> =
+            Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
+                .unwrap()
+                .into();
+        let hash = Ok(block.as_ref().into());
+        vec![(block, hash)]
+    });
+
+static INVALID_TIME_BLOCK_TRANSCRIPT: Lazy<Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>> =
+    Lazy::new(|| {
+        let mut block: Block =
+            Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
+                .unwrap();
+
+        // Modify the block's time
+        // Changing the block header also invalidates the header hashes, but
+        // those checks should be performed later in validation, because they
+        // are more expensive.
+        let three_hours_in_the_future = Utc::now()
+            .checked_add_signed(chrono::Duration::hours(3))
+            .ok_or("overflow when calculating 3 hours in the future")
+            .map_err(|e| eyre!(e))
+            .unwrap();
+        block.header.time = three_hours_in_the_future;
+
+        vec![(Arc::new(block), Err(TransError::Any))]
+    });
+
+static INVALID_HEADER_SOLUTION_TRANSCRIPT: Lazy<
+    Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>,
+> = Lazy::new(|| {
+    let mut block: Block =
+        Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..]).unwrap();
 
     // Change nonce to something invalid
     block.header.nonce = [0; 32];
 
-    let ready_verifier_service = block_verifier.ready_and().await.map_err(|e| eyre!(e))?;
-
-    // Error: invalid equihash solution for BlockHeader
-    ready_verifier_service
-        .call(Arc::new(block.clone()))
-        .await
-        .expect_err("expected the equihash solution to be invalid");
+    vec![(Arc::new(block), Err(TransError::Any))]
+});
+
+static INVALID_COINBASE_TRANSCRIPT: Lazy<Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>> =
+    Lazy::new(|| {
+        let header =
+            BlockHeader::zcash_deserialize(&zebra_test::vectors::DUMMY_HEADER[..]).unwrap();
+
+        // Test 1: Empty transaction
+        let block1 = Block {
+            header,
+            transactions: Vec::new(),
+        };
+
+        // Test 2: Transaction at first position is not coinbase
+        let mut transactions = Vec::new();
+        let tx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]).unwrap();
+        transactions.push(Arc::new(tx));
```

Comment:
Does this work instead?
```suggestion
        let trans = zebra_test::vectors::DUMMY_TX1.zcash_deserialize_into().unwrap();
        transactions.push(trans);
```

Reply:
Yes, something like this worked, added at https://github.com/ZcashFoundation/zebra/pull/675/commits/2ced9f7cf713b5c83637ece0874cf2630ebcc673